lib/cartopy/crs.py

```diff
@@ -821,6 +821,10 @@ def project_geometry(self, geometry, src_crs=None):
             raise TypeError('Source CRS must be an instance of CRS'
                             ' or one of its subclasses, or None.')
         geom_type = geometry.geom_type
+        if geom_type == 'GeometryCollection':
+            return sgeom.GeometryCollection(
+                [self.project_geometry(geom, src_crs) for geom in geometry.geoms])
```

Comment:
Perhaps add this to the `_method_map` and add the logic to a new function `_project_geometry_collection` to keep it consistent with the others.